icon/Icon.js

```diff
@@ -915,43 +925,66 @@ export const deserializeIcon = function(iconDef) {
  * @param {String} extension
  * @return {Element}
  */
-export const fileIcon = function(extension) {
+
+export const [FileIcon, fileIcon] = hoistCmp.withFactory({
```

Comment:
Think we can skip the component, since we have Icon.fileIcon()

Reply:
let's remove this component -- users can just use the elem factory in their render method.